### PR TITLE
Feature/saved section filtering

### PR DESCRIPTION
**Pull request by hvohr**
What I did: Added filter functionality to the saved section and a functioning search bar. Search bar searches only within the saved section. Clear button was also added that clears the search bar in the main section and the saved section.

Did this break anything?
- [x] No

Type of change
- [x] New feature (non-breaking change which adds functionality)

Checklist:
- [x]  The code runs locally
- [x]  The code is DRY. If it's not, I tried my best
- [x]  I reviewed my code before pushing

What's next? Fixing the delete functionality and adding the fetch API functionality

### CODE REVIEW

File: src/domUpdates.js [hunk 1]
```diff
@@ -169,7 +220,7 @@ function randomizeHomePage() {
 
 function renderFilteredRecipes() {
   const tags = Array.from(checkCategories).filter((category) => category.checked).map(c => c.id)
-  if (tags.length === 0) {
+  if (tags.length === 0 && savedRecipeDisplay.classList[1] === 'hidden') {
```
Comment (Denzel-Turambi) [inline src/domUpdates.js:223]: Good addition to the conditional!

**Review by Denzel-Turambi — APPROVED** [review]

File: dist/index.html [hunk 2]
```diff
@@ -12,7 +12,9 @@
     <h1 class = "title">What's Cookin?</h1>
     <nav class="navbar">
       <button class="all-recipes">All Recipes</button>
-      <input id = "search-bar" type="text" placeholder=" Search...">
+      <input class = "nav-search-bar" id = "search-bar" type="text" placeholder=" Search...">
+      <input class = "saved-search hidden" id = "saved-search-bar" type="text" placeholder=" Search Saved Recipes...">
```
Comment (andreasorensen) [inline dist/index.html:16]: I love that the user can search through their savedRecipes without looking for another search bar!

File: src/domUpdates.js [hunk 3]
```diff
@@ -14,24 +14,28 @@ const allRecipeDisplay = document.querySelector('.all-recipes-display');
...
 const checkCategories = document.getElementsByName('checkbox');
 const searchInput = document.getElementById('search-bar');
+const savedSearchInput = document.getElementById('saved-search-bar')
 const recipeTitle = document.querySelector('h2');
 const singleRecipeDisplay = document.querySelector('.single-recipe-display');
 const homeButton = document.querySelector('.title')
 const saveRecipeButton = document.querySelector('.save-recipe-button')
 const savedRecipesButton = document.querySelector('.saved-recipes')
 const savedRecipeDisplay = document.querySelector('.saved-recipe-display')
+const clearButton = document.querySelector('.clear-search-button')
```
Comment (andreasorensen) [inline src/domUpdates.js:24]: YES!

**Review by andreasorensen — APPROVED** [review]

**Review by fernandorobles97 — APPROVED** [review]
Great job on the new functionalities!

**Pull request merged**